docs/source/migrations/1.0.mdx

```diff
@@ -0,0 +1,455 @@
+      return nil
+    }
+
+    return CacheKeyInfo(id: id)
+  }
+}
+```
+
+Or using the [JSON value convenience initializer](./../caching/cache-key-resolution#json-value-convenience-initializer) like so:
+
+```swift
+public extension MySchema {
+  static func cacheKeyInfo(for type: Object, object: JSONObject) -> CacheKeyInfo? {
+    return try? CacheKeyInfo(jsonValue: object["id"])
+  }
+}
+```
+
+### Local Cache Mutations
+
+In the 0.x version, you could directly make changes to data in the local cache using any of your generated operation or fragment models. While the APIs for direct cache access have not change significantly, the generated model objects are now immutable by default. You can still read cache data directly using all of your generated models, but in order to mutate cache data, you will now need to define seperate [local cache mutation](./../caching/cache-transactions#defining-local-cache-mutations) operations or fragments.
+
+You can define a local cache mutation model by applying the `@apollo_client_ios_localCacheMutation` directive to any GraphQL operation or fragment definition.
+
+> For a detailed explanation of the new local cache mutation APIs, read the [direct cache access documentation](./../caching/cache-transactions).
+
+<SeperateLocalCacheMutationsNote />
+
+#### Example
+
+Given an operation and write transaction from the 0.x version:
+
+<CodeColumns>
+
+```graphql
+query UserDetails {
+  loggedInUser {
+    id
+    name
+    posts {
+      id
+      body
+    }
+  }
+}
+```
+
+```swift
+store.withinReadWriteTransaction({ transaction in
+  let cacheMutation = UserDetailsQuery()
+
+  let newPost = UserDetailsQuery.Data.LoggedInUser.Post(id: "789, body: "This is a new post!")
+
+  try transaction.update(cacheMutation) { (data: inout UserDetailsQuery.Data) in
+    data.loggedInUser.posts.append(newPost)
+  }
+})
+```
+</CodeColumns>
+
+You can migrate this to a new `LocalCacheMutation` like so:
+
+<CodeColumns>
+
+```graphql
+query AddUserPostLocalCacheMutation @apollo_client_ios_localCacheMutation {
+  loggedInUser {
+    posts {
+      id
+      body
+    }
+  }
+}
+```
+
+```swift
+store.withinReadWriteTransaction({ transaction in
+  let cacheMutation = AddUserPostLocalCacheMutation()
+
+  let newPost = AddUserPostLocalCacheMutation.Data.LoggedInUser.Post(data: DataDict(
+    ["__typename": "Post", "id": "789", "body": "This is a new post!"],
+    variables: nil
+  ))
+
+  try transaction.update(cacheMutation) { (data: inout AddUserPostLocalCacheMutation.Data) in
+    data.loggedInUser.posts.append(newPost)
+  }
+})
+```
+
+</CodeColumns>
+
+### Nullable Input Values
+
+According to [the GraphQL spec](http://spec.graphql.org/October2021/#sec-Null-Value), explicitly providing a `null` value for an input value to a field argument is semantically different from not providing a value at all (`nil`). This enum allows you to distinguish your input values between `null` and `nil`.
```

Comment:
The last sentence mentions "This enum" but hasn't introduced it yet. Might be better to simply remove that last sentence as the concept and our enum is explained in detail below.